corehq/apps/userreports/pillow.py

```diff
@@ -222,6 +225,101 @@ def _save_doc_to_table(self, domain, table, doc, eval_context):
             # remove it until the next bootstrap call
             self.table_adapters_by_domain[domain].remove(table)
 
+    def process_changes_chunk(self, pillow_instance, changes):
+        failed_changes = set()
+
+        def get_docs(_changes, domain):
+            # get docs in bulk and validate revision
+            if not _changes:
+                return []
+            # break up doctype
+            _changes_by_doc_store = defaultdict(list)
+            for change in _changes:
+                _changes_by_doc_store[change.document_store].append(change)
+
+            # query
+            docs = []
+            for doc_store, d_changes in six.iteritems(_changes_by_doc_store):
+                docs.extend(list(doc_store.iter_documents([change.id for change in d_changes])))
```

Comment:
I think this is fine for doc sizes of 100, but if we ever went larger we probably wouldn't want it as a list

Reply:
Do you think it would be anymore orders of magnitude?